pkg/healthcheck/healthcheck.go

```diff
@@ -2263,21 +2265,25 @@ func checkResources(resourceName string, objects []runtime.Object, expectedNames
 // Check if there's a pod with the "opaque ports" annotation defined but a
 // service selecting the aforementioned pod doesn't define it
 func (hc *HealthChecker) checkMisconfiguredOpaquePortAnnotations(ctx context.Context) error {
-	services, err := hc.GetServices(ctx)
+	// Initialize and sync the kubernetes API
+	// This is used instead of `hc.kubeAPI` to limit multiple k8s API requests
+	// and use the caching logic in the shared informers
+	kubeAPI := controllerK8s.NewAPI(hc.kubeAPI, nil, nil, controllerK8s.Endpoint, controllerK8s.Pod, controllerK8s.Svc)
```

Comment:
We may want to add a TODO here for factoring the shared informer code out of the `controller/` package.  Having `pkg` depend on `controller` feels like a backwards dependency and may lead to circular dependencies in the future.

Reply:
Would we be able to move the healtchecker to use the `controller` package's k8s API instead of it's current `k8s` package k8s API? We could still refactor so that it's not part of the `controller` package, but it would be nice if we could use the informers everywhere in the healthchecker instead of just here

Reply:
Added the TODO like you suggested! The informer discussion seems to be a bigger design decision for now! 🤔